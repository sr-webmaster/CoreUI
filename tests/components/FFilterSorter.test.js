import { mount } from '@vue/test-utils'
import Component from '@freshinup/core-ui/src/components/FFilterSorter.vue'

describe('Filter Sorter', () => {
  describe('Snapshots', () => {
    test('default', () => {
      const wrapper = mount(Component)
      expect(wrapper.element).toMatchSnapshot()
    })

    test('Slot expanded', async () => {
      const wrapper = mount(Component, {
        slots: {
          expanded: '<div class="slot" />'
        }
      })
      expect(wrapper.find('.v-btn__content').text()).toContain('More Filters')
      wrapper.vm.toggleExpanded()
      await wrapper.vm.$nextTick()
      expect(wrapper.find('.v-btn__content').text()).toContain('Less Filters')
    })
  })
  describe('Properties', () => {
    describe('flat-container', () => {
      test('defaults to being raised (the default for V-Cards)', () => {
        const wrapper = mount(Component)
        expect(wrapper.classes('v-card')).toEqual(true)
        expect(wrapper.classes('v-card--flat')).toEqual(false)
      })
      test('when set will place the v-card--flat', () => {
        const wrapper = mount(Component, {
          propsData: {
            flatContainer: true
          }
        })
        expect(wrapper.classes('v-card')).toEqual(true)
        expect(wrapper.classes('v-card--flat')).toEqual(true)
      })
    })
  })
  describe('Watch', () => {
    test('expanded', async () => {
      const wrapper = mount(Component, {
        propsData: {
          expanded: false
        }
      })
      wrapper.setProps({ expanded: true })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.isExpanded).toEqual(true)
    })
  })
  describe('Methods', () => {
    test('clear', () => {
      const wrapper = mount(Component)
      wrapper.vm.term = 'something'
      wrapper.vm.clear()
      expect(wrapper.vm.term).toEqual('')
      expect(wrapper.emitted('clear')).toBeTruthy()
    })
    test('collapse', () => {
      const wrapper = mount(Component)
      wrapper.vm.isExpanded = true
      wrapper.vm.collapse()
      expect(wrapper.vm.isExpanded).toEqual(false)
      expect(wrapper.emitted('collapsed')).toBeTruthy()
    })
    test('expand', () => {
      const wrapper = mount(Component)
      wrapper.vm.isExpanded = false
      wrapper.vm.expand()
      expect(wrapper.vm.isExpanded).toEqual(true)
      expect(wrapper.emitted('expanded')).toBeTruthy()
    })
    test('toggleExpanded', () => {
      const wrapper = mount(Component)
      wrapper.vm.isExpanded = false
      wrapper.vm.toggleExpanded()
      expect(wrapper.vm.isExpanded).toEqual(true)
      expect(wrapper.emitted('expanded')).toBeTruthy()
      wrapper.vm.toggleExpanded()
      expect(wrapper.vm.isExpanded).toEqual(false)
      expect(wrapper.emitted('collapsed')).toBeTruthy()
    })
    test('getRunParams', () => {
      const wrapper = mount(Component)
      wrapper.vm.term = 'some'
      wrapper.vm.sort = 'thing'
      expect(wrapper.vm.getRunParams()).toEqual({
        term: 'some',
        orderBy: 'thing'
      })
    })
    test('run', done => {
      const wrapper = mount(Component)
      wrapper.vm.term = 'some'
      wrapper.vm.sort = 'thing'
      wrapper.vm.run()
      setTimeout(() => {
        expect(wrapper.emitted('run')).toBeTruthy()
        expect(wrapper.emitted('run')[0][0]).toEqual({
          term: 'some',
          orderBy: 'thing'
        })
        done()
      }, 501)
    })
    test('termChange', () => {
      const wrapper = mount(Component)
      wrapper.vm.termChange('something')
      expect(wrapper.vm.term).toEqual('something')
    })
  })
  describe('CSS Names', () => {
    test('follows FreshinUp BEM standards', () => {
      const wrapper = mount(Component)
      expect(wrapper.classes('f-filter-sorter')).toEqual(true)
      expect(wrapper.exists('.f-filter-sorter__main-layout')).toEqual(true)
      expect(wrapper.exists('.f-filter-sorter__sortable-select')).toEqual(true)
    })
  })
})
