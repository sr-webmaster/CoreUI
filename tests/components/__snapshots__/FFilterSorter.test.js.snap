// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Sorter Snapshots default 1`] = `
<div
  class="f-filter-sorter v-card v-sheet theme--light"
>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container fluid"
    >
       
      <div
        class="layout f-filter-sorter__main-layout row wrap"
      >
        <div
          class="flex shrink"
        >
          <div
            class="v-card__text"
          >
            Filter:
          </div>
        </div>
         
        <div
          class="flex grow"
        >
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <input
                    placeholder="Start typing to Search"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <!---->
         
        <!---->
         
      </div>
       
      <!---->
       
      <div
        class="layout f-filter-sorter__expanded-layout align-center justify-center row fill-height"
        style="display: none;"
      >
        <div
          class="flex"
        />
         
        <div
          class="flex text-no-wrap"
        >
          <button
            class="v-btn v-btn--large theme--light undefined-text--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Clear Filters
          
            </div>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
